sdk/core/Azure.Core/tests/common/ModelReaderWriter/Models/DiscriminatorSet/BaseModel.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.ClientModel;
+using System.Net.ClientModel.Core;
+using System.Net.ClientModel.Internal;
+using System.Text.Json;
+
+namespace Azure.Core.Tests.ModelReaderWriterTests.Models
+{
+    [ModelReaderProxy(typeof(UnknownBaseModel))]
+    public abstract class BaseModel : IUtf8JsonSerializable, IJsonModel<BaseModel>
+    {
+        private Dictionary<string, BinaryData> _rawData;
+
+        public static implicit operator RequestContent(BaseModel baseModel)
+        {
+            if (baseModel == null)
+            {
+                return null;
+            }
+
+            return RequestContent.Create(PipelineContent.CreateContent(baseModel, ModelReaderWriterOptions.DefaultWireOptions));
```

Comment:
I think this should be able to call a RequestContent.Create(IModel) method directly as well rather than creating a PipelineContent first.  This will simplify the pattern in the generated code.